### PR TITLE
Arrange, restructure code

### PR DESCRIPTION
**Pull request by MauNguyenVan**
Please review my code change. Thanks

### CODE REVIEW

File: Source/HelixToolkit.Maths/Matrix4x4Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace HelixToolkit.Maths
+{
+    public static class Matrix4x4Extensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix4x4 Inverted(this Matrix4x4 m)
+        {
+            return Matrix4x4.Invert(m, out var result) ? result : Matrix4x4.Identity;
```
Comment (holance) [inline Source/HelixToolkit.Maths/Matrix4x4Extensions.cs:14]: We probably don't want a silent failure if the matrix is not able to be inverted.

Comment (holance) [reply Source/HelixToolkit.Maths/Matrix4x4Extensions.cs:14]: Return identity matrix is still a bug. We can log the error if it's not able to be inverted.

Comment (holance) [reply Source/HelixToolkit.Maths/Matrix4x4Extensions.cs:14]: You can return either identity or zeros, just log the error.

File: Source/HelixToolkit.Maths/Plane.cs [hunk 2]
```diff
@@ -341,13 +341,14 @@ public static bool IntersectsLine(ref Plane p, ref Vector3 p0, ref Vector3 p1, o
...
             if (d == 0)
             {
                 intersection = Vector3.Zero;
                 return false;
             }
-            float u = (Vector3.Dot(p0, p.Normal) + p.D) / d;
+            float u = (Vector3.Dot(p0, planeNormalize) + p.D) / d;
             intersection = p0 + u * (p1 - p0);
             return true;
         }
```
[the remark this answers is not in the document]
Comment (holance) [reply Source/HelixToolkit.Maths/Plane.cs:354]: Usually normal means it's already normalized.

File: Source/HelixToolkit.Maths/Vector3.cs [hunk 3]
```diff
@@ -148,6 +144,19 @@ public static bool IsZero(this Vector3 v)
...
 
+        /// <summary>
+        /// Determines whether the specified vector is undefined (NaN,NaN,NaN).
+        /// </summary>
+        /// <param name="v">The vector.</param>
+        /// <returns>
+        /// <c>true</c> if the specified vector is undefined; otherwise, <c>false</c>.
+        /// </returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsUndefined(this Vector3 v)
```
Comment (ArieRiezebos) [inline Source/HelixToolkit.Maths/Vector3.cs:155]: Comment suggests that it returns true if all (X, Y, Z) are NaN, but it returns true if any of them is NaN. I'm also wondering what the difference is with method AnyUndefined() on line 879. What it maybe the intention to use && for this method and || for the other?

Comment (holance) [reply Source/HelixToolkit.Maths/Vector3.cs:155]: Comment is wrong

**Review by holance — APPROVED** [review]

**Review by holance — APPROVED** [review]

**Pull request merged**